engine/src/cubos/engine/renderer/plugin.cpp

```diff
@@ -169,17 +136,17 @@ static void draw(Write<Renderer> renderer, Read<ActiveCameras> activeCameras, Wr
 
         if (auto components = query[activeCameras->entities[i]])
         {
-            auto [localToWorld, camera] = *components;
+            auto [localToWorld, camera, viewport] = *components;
             cameras[cameraCount].fovY = camera->fovY;
             cameras[cameraCount].zNear = camera->zNear;
             cameras[cameraCount].zFar = camera->zFar;
+            viewports[i].position = {viewport->positionX, viewport->positionY};
+            viewports[i].size = {viewport->sizeX, viewport->sizeY};
```

Comment:
By changing the viewport in the query to `OptRead`, we here should check it:
```suggestion
            if (viewport)
            {
                viewports[i].position = {viewport->positionX, viewport->positionY};
                viewports[i].size = {viewport->sizeX, viewport->sizeY};
           }
           else
           {
                // set the viewport to the entire screen by default
           }
```